playpen/all_27.sh

```diff
@@ -1,22 +1,52 @@
 #!/bin/bash
+# Downgrade a Pulp system from master to 2.7.
+#
+# WARNING: This script drops a mongo database, uninstalls software, and
+# performs other destructive actions.
+source ~/.bashrc  # get pstop, pstart, etc
+set -euo pipefail
 
-#  When using this script, you will need to fix your database accordingly. If moving backward, you
-#  will need to drop the database, and you will always need to run the migrations.
+pstop
 
-
-# pulp_docker, pulp_python, pulp_ostree excluded because they are versioned differently
-for repo in pulp pulp_puppet pulp_rpm
-do
-    pushd /home/vagrant/devel/$repo
-    git checkout 2.7-dev
-    find ~/devel -name "*.py[c0]" -delete
-    sudo ./pulp-dev.py --install
-    popd
+# Downgrade these plugins to version 2.7.
+pushd ~/devel
+for repo in pulp{,_puppet,_rpm}; do
+    if [ -d "$repo" ]; then
+        pushd "$repo"
+        git checkout 2.7-dev
+        find . -name '*.py[c0]' -delete
+        sudo ./pulp-dev.py --install
+        popd
+    fi
 done
+popd
 
-for repo in pulp_docker pulp_python
-do
-    pushd /home/vagrant/devel/$repo
-    sudo ./pulp-dev.py --uninstall
-    popd
+# Completely uninstall these plugins. (Why?)
+pushd ~/devel
+for repo in pulp_{docker,python}; do
+    if [ -d "$repo" ]; then
+        pushd "$repo"
+        sudo ./pulp-dev.py --uninstall
+        popd
+    fi
 done
+popd
+
+set -x
+
+# These aren't used by Pulp 2.7.
+sudo rm /etc/httpd/conf.d/pulp_content.conf
+sudo rm /etc/httpd/conf.d/pulp_streamer.conf
+
+# Pulp 2.7 is incompatible with python-mongoengine >= 3 and python-pymongo >=
+# 0.10.
+sudo dnf  -y remove python-{mongoengine,pymongo}
+sudo dnf config-manager --disablerepo pulp-nightlies
+sudo dnf config-manager --enablerepo pulp-2.7-beta
+sudo dnf -y install python-mongoengine-0.8.8 python-pymongo-2.5.2
+
+# Drop and recreate database.
+mongo pulp_database --eval 'db.dropDatabase()'
```

Comment:
I removed this from the original script because I would rather an extra step than anyone accidentally losing data.